app/scripts/lib/multichain-api/caip25permissions.ts

```diff
@@ -61,3 +66,69 @@ export const caip25EndowmentBuilder = Object.freeze({
   targetName: Caip25EndowmentPermissionName,
   specificationBuilder,
 } as const);
+
+/**
+ * Factories that construct caveat mutator functions that are passed to
+ * PermissionController.updatePermissionsByCaveat.
+ */
+export const Caip25CaveatMutatorFactories = {
+  [Caip25CaveatType]: {
+    removeScope,
+  },
+};
+
+
+const reduceKeysHelper = (acc, [key, value]) => {
+  return {
+    ...acc,
+    [key]: value,
+  };
+};
+
+/**
+ * Removes the target account from the value arrays of all
+ * `endowment:caip25` caveats. No-ops if the target scopeString is not in
+ * the existing scopes,.
+ *
+ * @param {string} targetScopeString - The address of the account to remove from
+ * all accounts permissions.
+ * @param {ScopeParamsObject} existingScopeParams - The account address array from the
+ * account permissions.
+ */
+export function removeScope(targetScopeString, existingScopes) {
+  const newRequiredScopes = Object.entries(
+    existingScopes.requiredScopes,
+  ).filter(([scope]) => scope !== targetScopeString);
+  const newOptionalScopes = Object.entries(
+    existingScopes.optionalScopes,
+  ).filter(([scope]) => {
+    return scope !== targetScopeString;
+  });
+
+  const requiredScopesRemoved =
+    newRequiredScopes.length !== Object.entries(existingScopes.requiredScopes).length;
+  const optionalScopesRemoved =
+    newOptionalScopes.length !== Object.entries(existingScopes.optionalScopes).length;
+
+  if (requiredScopesRemoved) {
```

Comment:
this can probably just be a check for `existingScopes.requiredScopes[targetScopeString]`? nit i suppose

Reply:
ill leave this in in case we need to support removing multiple scopes at some point